lib/Backend/PreLowerPeeps.cpp

```diff
@@ -98,6 +98,14 @@ IR::Instr *Lowerer::PeepShl(IR::Instr *instrShl)
     instrDef->Remove();
 
     IntConstType oldValue = src2->AsIntConstOpnd()->GetValue();
+
+    // Left shift operator (<<) on arm32 is implemented by LSL which doesn't discard bits beyond lowerest 5-bit.
```

Comment:
Louis is right. Both LSL on ARM and SHL on x86 accept shift count operand in register which could be 32-bit value at least. This code is handling the different behavior when the register has value larger than 32 (shift more than 32 bits).

Will remove the #ifdef.

---
In reply to: [100138094](https://github.com/Microsoft/ChakraCore/pull/2486#discussion_r100138094) [](ancestors = 100138094,100117486)